nimbus-core/src/main/java/com/antheminc/oss/nimbus/converter/excel/UnivocityExcelToCSVConverter.java

```diff
@@ -0,0 +1,127 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.antheminc.oss.nimbus.converter.excel;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.poi.EncryptedDocumentException;
+import org.apache.poi.openxml4j.exceptions.InvalidFormatException;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.DataFormatter;
+import org.apache.poi.ss.usermodel.FormulaEvaluator;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+
+import com.antheminc.oss.nimbus.FrameworkRuntimeException;
+import com.univocity.parsers.csv.CsvWriter;
+import com.univocity.parsers.csv.CsvWriterSettings;
+
+/**
+ * <p>Uses Univocity to convert excel files to their .csv equivalent and stores
+ * that file in the local filesystem's temporary storage.
+ * 
+ * @author Tony Lopez
+ * @author Sandeep Mantha
+ */
+public class UnivocityExcelToCSVConverter implements ExcelToCsvConverter {
+
+	@Override
+	public List<File> convert(InputStream inputStream, ExcelParserSettings settings) throws IOException {
+		try (Workbook workbook = WorkbookFactory.create(inputStream)) {
+			return convert(workbook, settings);
+		} catch (EncryptedDocumentException | InvalidFormatException e) {
+			throw new FrameworkRuntimeException("An error occurred while conerting the Excel workbook object.", e);
+		}
+	}
+
+	private List<File> convert(Workbook workbook, ExcelParserSettings excelParserSettings)
+			throws IOException {
+		FormulaEvaluator evaluator = workbook.getCreationHelper().createFormulaEvaluator();
+		DataFormatter formatter = new DataFormatter(true);
+
+		int[] sheetNumbers = excelParserSettings.getSheetNumbersToParse();
+		if (excelParserSettings.isParseFirstSheetOnly()) {
+			sheetNumbers = new int[] { 0 };
+		} else if (excelParserSettings.isParseAllSheets()) {
+			sheetNumbers = ArrayUtils.toPrimitive(IntStream.rangeClosed(0, workbook.getNumberOfSheets() - 1)
+				    .boxed().toArray(Integer[]::new));
+		}
+		
+		List <File> csvFiles = new ArrayList<>();
+		for(int sheetNumber : sheetNumbers) {
+			Sheet sheet = workbook.getSheetAt(sheetNumber);
+			File csvFile = parseSheet(sheet, formatter, evaluator);
+			csvFiles.add(csvFile);
+		}
+		return csvFiles;
+	}
+	
+	private File parseSheet(Sheet sheet, DataFormatter formatter, FormulaEvaluator evaluator) throws IOException {
+		Iterator<Row> rowIterator = sheet.iterator();
+
+		CsvWriterSettings settings = new CsvWriterSettings();
```

Comment:
Can the settings be injected as a bean config?

Reply:
I think a spring bean is a little bit overkill, but I will put it as a constructor argument to the `UnivocityExcelToCSVConverter` class and inject it during it's bean creation.

Reply:
Fixed.